app/controllers/application_controller.rb

```diff
@@ -2,4 +2,10 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  
+  before_filter :set_current_user
+  
+  def set_current_user
```

Comment:
Right idea.. instead define a `current_customer` method that returns a customer. Then you'll need the association to get to Profile elsewhere.